### PR TITLE
scripts: Add initial codegen scripts

### PR DESCRIPTION
**Pull request by spencer-lunarg**
For https://github.com/KhronosGroup/Vulkan-Utility-Libraries/issues/13

I didn't run and check in the code as need to change where they will go

`vulkan_object.py` is the same as VVL, `base_generator.py` was improved and going to port changes back

### CODE REVIEW

File: scripts/generate_source.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    from generators.base_generator import BaseGeneratorOptions
+    from generators.dispatch_table_helper_generator import DispatchTableHelperOutputGenerator
+    from generators.layer_dispatch_table_generator import LayerDispatchTableOutputGenerator
+    from generators.enum_string_helper_generator import EnumStringHelperOutputGenerator
+
+    # Build up a list of all generators and custom options
+    generators = {
+        'vk_dispatch_table_helper.h' : {
+            'generator' : DispatchTableHelperOutputGenerator,
+            'directory' : 'include/vulkan/layer',
```
Comment (spencer-lunarg) [inline scripts/generate_source.py:42]: @juan-lunarg this is where we can simply change the location

**Review by juan-lunarg — APPROVED** [review]

File: scripts/generate_source.py (hunk 1 above)
Comment (juan-lunarg) [reply scripts/generate_source.py:42]: Gotcha 👍🏾

Comment (spencer-lunarg) [reply scripts/generate_source.py:42]: @juan-lunarg  Do you know where you want these, then we can set them, run the script and check in the generated code

Comment (juan-lunarg) [reply scripts/generate_source.py:42]: Since these are all header only they should go in the `include/vulkan` directory.

Comment (juan-lunarg) [reply scripts/generate_source.py:42]: FYI @mikes-lunarg

**Pull request merged**